### PR TITLE
fix model_name exception for turbomind

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

fix model_name exception for turbomind

## Modification

`_build_turbomind`

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
ping @lvhan028 @irexyc

**Comment by zhyncs** [comment]
ref https://github.com/InternLM/lmdeploy/pull/1085

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -123,6 +124,12 @@ def _build_turbomind(
         assert isinstance(backend_config, TurbomindEngineConfig), 'Please'\
             ' use TurbomindEngineConfig imported from lmdeploy.messages for ' \
             'turbomind backend'
+        from lmdeploy import turbomind as tm
+        self.engine = tm.TurboMind.from_pretrained(
```
Comment (lvhan028) [inline lmdeploy/serve/async_engine.py:128]: If it is moved here, the default `session_len` cannot be updated to `backend_config`

Comment (zhyncs) [reply lmdeploy/serve/async_engine.py:128]: > If it is moved here, the default `session_len` cannot be updated to `backend_config`

Let me have a look

**Comment by lvhan028** [comment]
I am gonna go with PR #1126

**Comment by irexyc** [comment]
We want to unify the initialization process of pytorch turbomind engine and remove model name(chat template) from engine initializaton in the future. Therefore, we shoud handle it in pipeline api.

**Pull request closed without merging**